src/query/frontend/semantic/symbol_generator.cpp

```diff
@@ -470,6 +501,20 @@ bool SymbolGenerator::PostVisit(Exists & /*exists*/) {
   return true;
 }
 
+bool SymbolGenerator::PreVisit(NamedExpression &named_expression) {
+  auto &scope = scopes_.back();
+
+  if (scope.in_call_subquery && scope.in_return) {
+    auto *ident = utils::Downcast<Identifier>(named_expression.expression_);
+
+    if (!ident && !named_expression.is_aliased_) {
+      throw SemanticException("Expression returned from subquery must be aliased (use AS)!");
+    }
+  }
+
+  return true;
+}
```

Comment:
```suggestion
bool SymbolGenerator::PreVisit(NamedExpression &named_expression) {
    if (auto &scope = scopes_.back(); scope.in_call_subquery && scope.in_return &&
      !utils::Downcast<Identifier>(named_expression.expression_) && !named_expression.is_aliased_)
  {
      throw SemanticException("Expression returned from subquery must be aliased (use AS)!");
  }
  return true;
}
```
Feel free to modify it but I think is "wrong" to have two scopes for and conditions